tests/DependencyInjection/ConfigurationTest.php

```diff
@@ -60,7 +60,24 @@ public function testInvalidNativePageRange(mixed $range): void
         );
     }
 
-    public function testWithExtraHeadersConfiguration(): void
+    /**
+     * @return iterable<string, array<array-key, array<string, string>|list<array<string, string>>>>
```

Comment:
```suggestion
     * @return iterable<string, list<array<array-key, string>|list<array{name: array-key, value: string}>>>
```

I think this should be good.